packages/gafl-webapp-service/src/pages/concessions/date-of-birth/__tests__/route.spec.js

```diff
@@ -1,31 +1,73 @@
 import { getData, validator } from '../route'
 import pageRoute from '../../../../routes/page-route.js'
 import { nextPage } from '../../../../routes/next-page.js'
+import { LICENCE_FOR } from '../../../../uri.js'
+import GetDataRedirect from '../../../../handlers/get-data-redirect.js'
 
 jest.mock('../../../../routes/next-page.js', () => ({
   nextPage: jest.fn()
 }))
 jest.mock('../../../../routes/page-route.js')
 
 describe('name > route', () => {
-  const mockTransactionCacheGet = jest.fn()
-
-  const mockRequest = {
+  const mockRequest = (statusGet = () => {}, transactionGet = () => {}) => ({
     cache: () => ({
       helpers: {
         transaction: {
-          getCurrentPermission: mockTransactionCacheGet
+          getCurrentPermission: transactionGet
+        },
+        status: {
+          getCurrentPermission: statusGet
         }
       }
     })
-  }
+  })
 
   describe('getData', () => {
     it('should return isLicenceForYou as true, if isLicenceForYou is true on the transaction cache', async () => {
-      mockTransactionCacheGet.mockImplementationOnce(() => ({ isLicenceForYou: true }))
-      const result = await getData(mockRequest)
+      const transaction = () => ({
+        isLicenceForYou: true
+      })
+      const status = () => ({
+        [LICENCE_FOR.page]: true
+      })
+      const result = await getData(mockRequest(status, transaction))
       expect(result.isLicenceForYou).toBeTruthy()
     })
+
+    it('should return isLicenceForYou as false, if isLicenceForYou is false on the transaction cache', async () => {
+      const transaction = () => ({
+        isLicenceForYou: false
+      })
+      const status = () => ({
+        [LICENCE_FOR.page]: true
+      })
+      const result = await getData(mockRequest(status, transaction))
+      expect(result.isLicenceForYou).toBeFalsy()
+    })
+  })
+
+  describe('redirectToStartOfJourney', () => {
+    it('should throw a redirect if not been to LICENCE_FOR page', async () => {
+      const transaction = () => ({
+        isLicenceForYou: true
+      })
+      const status = () => ({
+        [LICENCE_FOR.page]: false
+      })
+      const func = async () => await getData(mockRequest(status, transaction))
+      await expect(func).rejects.toThrow(new GetDataRedirect(LICENCE_FOR.uri))
```

Comment:
Until I get around to adding that new matcher, this would work just as well if you did:
```
await expect(func).rejects.toThrow(GetDataRedirect)
```
I'll leave it up to you whether you change it, just don't assume that you're testing that the thrown error has `redirectUrl` set to `LICENCE_FOR.uri`. You could change `redirectToStartOfJourney` to be:
```
const redirectToStartOfJourney = status => {
  if (!status[LICENCE_FOR.page]) {
     throw new GetDataRedirect('any old rubbish')
  }
 }
```
and your test would still pass.